src/EventListener/InvalidateRequestSubscriber.php

```diff
@@ -37,8 +44,48 @@ public static function getSubscribedEvents() : array
      */
     public function invalidate(InvalidateRequestEvent $event)
     {
-        $cacheStorage = $this->cacheMiddleware->getCacheStorage();
-        $requests     = $event->getRequests();
-        array_walk($requests, [$cacheStorage, 'delete']);
+        $client = $event->getClient();
+        $objectIdentifier = $this->getClientIdentifier($client);
+
+        if (isset($this->cacheMiddlewares[$objectIdentifier])) {
+            $cacheStorage = $this->cacheMiddlewares[$objectIdentifier]->getCacheStorage();
+
+            $request = $this->buildRequest(
+                $event->getMethod(),
+                $client->getConfig('base_uri'),
+                $event->getUri()
+            );
+
+            $cacheStorage->delete($request);
+        }
+    }
+
+    /**
+     * @param string $method
+     * @param UriInterface|null $baseUri
+     * @param string $requestUri
+     *
+     * @return Request
+     */
+    protected function buildRequest(string $method, $baseUri, string $requestUri) : Request
+    {
+        if ($baseUri instanceof UriInterface) {
+            $uri = Psr7\UriResolver::resolve($baseUri, Psr7\uri_for($requestUri));
+        } else {
+            $uri = $requestUri;
+        }
+
+        return new Request($method, $uri);
+    }
+
+    /**
+     * @param Client $client
+     *
+     * @return string
+     */
+    protected function getClientIdentifier(Client $client) : string
+    {
+        return spl_object_hash($client);
```

Comment:
Thanks! Thta was my problem. I wanted to find a "proper" way to identify a GuzzleClient...